common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -191,30 +124,137 @@ foreach ($entry in $manifest) {
+  )
+
+  Start-Job -Name $entry.Name -ScriptBlock {
+    &"$using:repoRoot/eng/common/TestResources/New-TestResources.ps1" `
+       -BaseName  $using:baseName `
+       -ResourceGroupName $using:resourceGroupName `
+       -ServiceDirectory $using:entry.ResourcesDirectory `
+       -TestApplicationId $using:TestApplicationId `
+       -TestApplicationSecret $using:TestApplicationSecret `
+       -ProvisionerApplicationId $using:TestApplicationId `
+       -ProvisionerApplicationSecret $using:TestApplicationSecret `
+       -TestApplicationOid $using:TestApplicationOid `
+       -TenantId $using:TenantId `
+       -SubscriptionId $using:SubscriptionId `
+       -Location $using:Location `
+       -Environment $using:Environment `
+       -AdditionalParameters $using:AdditionalParameters `
+       -DeleteAfterHours 24 `
+       -Force `
+       -Verbose `
+       -CI:$using:CI
+  }
+}
+
+function Deploy-TestResources {
+  Param([Parameter(Mandatory = $true)] $deployManifest)
+
+  $deployedServiceDirectories = @{ }
+  $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+  $resourceGroupName = "rg-smoke-$baseName"
+  $dependencies = New-Object 'System.Collections.Generic.Dictionary[string,string]'
+  $runManifest = @()
+
+  # Use the same resource group name that New-TestResources.ps1 generates
+  Set-EnvironmentVariable -Name 'AZURE_RESOURCEGROUP_NAME' -Value $resourceGroupName
+
+  $entryDeployJobs = @()
+
+  try {
+    foreach ($entry in $deployManifest) {
+      if (!(Get-ChildItem -Path "$repoRoot/sdk/$($entry.ResourcesDirectory)" -Filter test-resources.json -Recurse)) {
+        Write-Verbose "Skipping $($entry.ResourcesDirectory): could not find test-resources.json"
+        continue
+      }
+
+      if ($deployedServiceDirectories.ContainsKey($entry.ResourcesDirectory) -ne $true) {
+        Write-Verbose "Starting deploy job for $($entry.ResourcesDirectory)"
+        $job = Start-NewTestResourcesJob $entry $baseName $resourceGroupName
+        $entryDeployJobs += $job
+        $deployedServiceDirectories[$entry.ResourcesDirectory] = $true;
+      }
+      else {
+        Write-Verbose "Skipping resource directory deployment (already deployed) for $($entry.ResourcesDirectory)"
+      }
 
+      Update-SamplesForService $entry
+      Update-SampleDependencies $entry $dependencies
+      $runManifest += $entry
+    }
+
+    Write-Verbose "Waiting for all deploy jobs to finish (will timeout after 15 minutes)..."
+    $entryDeployJobs | Wait-Job -TimeoutSec (15*60)
+    if ($entryDeployJobs | Where-Object {$_.State -eq "Running"}) {
+      $entryDeployJobs
+      throw "Timed out waiting for deploy jobs to finish:"
+    }
+
+    foreach ($job in $entryDeployJobs) {
+      if ($job.State -eq [System.Management.Automation.JobState]::Failed) {
+        $errorMsg = $job.ChildJobs[0].JobStateInfo.Reason.Message
+        LogWarning "Failed to deploy $($job.Name): $($errorMsg)"
+        Write-Warning "Failed to deploy $($job.Name)"
```

Comment:
What the double warning output?

Reply:
Was copied over from the original script. You're right it's redundant.